core/services/pipeline/orm.go

```diff
@@ -271,10 +271,27 @@ func (o *orm) InsertFinishedRun(run *Run, saveSuccessfulTaskRuns bool, qopts ...
 	return errors.Wrap(err, "InsertFinishedRun failed")
 }
 
+// DeleteRunsOlderThan deletes all pipeline_runs that have been finished for a certain threshold to free DB space
 func (o *orm) DeleteRunsOlderThan(ctx context.Context, threshold time.Duration) error {
 	q := o.q.WithOpts(pg.WithParentCtx(ctx))
-	err := q.ExecQ(`DELETE FROM pipeline_runs WHERE finished_at < $1`, time.Now().Add(-threshold))
-	return errors.Wrap(err, "DeleteRunsOlderThan failed")
+
+	err := pg.Batch(func(_, limit uint) (count uint, err error) {
+		result, err := q.Queryer.Exec(`DELETE FROM pipeline_runs WHERE finished_at < $1 LIMIT $2`, time.Now().Add(-threshold), limit)
```

Comment:
Is it important to control the order here? All else being equal, it seems preferable to delete oldest first and move forward in time.

Reply:
yeah, I think it would be preferable that way, made the change to do it by created_at ASC.